src/runtime_src/core/common/device.cpp

```diff
@@ -62,18 +62,77 @@ get_xclbin_uuid() const
 
 void
 device::
-register_axlf(const axlf* top)
+get_section_info(const char *buff, axlf_section_kind kind, 
+        const char *&sect_info, size_t *sect_size)
+{
+  struct connectivity       *group_conn = nullptr;
+  struct mem_topology       *group_topo = nullptr;
+
+  if (!buff)
+    return;
+
+  /* Typically input buff contains information regading group_topology and 
+   * group_connectivity with the following order :-
+   * buff:
+   *   1. group_topology
+   *   2. group_connectivity 
+   *
+   * Hence, extract information should be in the same order. 
+   */
+  group_topo = (struct mem_topology *)buff;
+  if (!group_topo)
+    throw std::runtime_error("failed to get memory topology information");
+
+  auto group_topo_size = sizeof(struct mem_topology) + 
+                            ((group_topo->m_count - 1) * sizeof(group_topo->m_mem_data)); 
+  buff += group_topo_size; 
+  group_conn = (struct connectivity *)buff;
+  if (!group_conn)
+    throw std::runtime_error("failed to get connectivity information");
+
+  auto mem_conn_size = sizeof(struct connectivity) + 
+                        ((group_conn->m_count - 1) * sizeof(group_conn->m_connection));
+                        
+  if (kind == MEM_TOPOLOGY) {
+    sect_info = (const char *)group_topo;
+    *sect_size = group_topo_size;
+  }
+  else if (kind == CONNECTIVITY) {
+    sect_info = (const char *)group_conn;
+    *sect_size =  mem_conn_size;
+  }
+}
+
+void
+device::
+register_axlf(const axlf* top, const char *info_buff)
 {
   m_axlf_sections.clear();
   m_xclbin_uuid = uuid(top->m_header.uuid);
   axlf_section_kind kinds[] = {EMBEDDED_METADATA, AIE_METADATA, IP_LAYOUT, CONNECTIVITY, MEM_TOPOLOGY, DEBUG_IP_LAYOUT, SYSTEM_METADATA};
```

Comment:
Please expand to include GROUP_TOPOLOGY and GROUP_CONNECTIVITY.  The conditional (if kind) code should chekc for if kind is GROUP_TOPOLOGY or GROUP_CONNECTIVITY and these are not in xclbin, in which case they should just be copies of MEM_TOPOLOGY and CONNECTIVITY

Reply:
Updated the function.